src/view/screens/ProfileFeed.tsx

```diff
@@ -335,44 +332,47 @@ export function ProfileFeedScreenInner({
             : undefined
         }
         avatarType="algo">
-        {feedInfo && hasSession && (
-          <>
-            <Button
-              disabled={isSavePending || isRemovePending}
-              type="default"
-              label={isSaved ? 'Unsave' : 'Save'}
-              onPress={onToggleSaved}
-              style={styles.btn}
-            />
-            <Button
-              testID={isPinned ? 'unpinBtn' : 'pinBtn'}
-              disabled={isPinPending || isUnpinPending}
-              type={isPinned ? 'default' : 'inverted'}
-              label={isPinned ? 'Unpin' : 'Pin to home'}
-              onPress={onTogglePinned}
-              style={styles.btn}
-            />
-          </>
-        )}
-        <NativeDropdown
-          testID="headerDropdownBtn"
-          items={dropdownItems}
-          accessibilityLabel={_(msg`More options`)}
-          accessibilityHint="">
-          <View style={[pal.viewLight, styles.btn]}>
-            <FontAwesomeIcon
-              icon="ellipsis"
-              size={20}
-              color={pal.colors.text}
-            />
-          </View>
-        </NativeDropdown>
+        <Box row gap='s' jce>
+          {feedInfo && hasSession && (
+            <>
+              <Button
+                size='small'
+                disabled={isSavePending || isRemovePending}
+                type="primary"
+                onPress={onToggleSaved}
+              >{isSaved ? 'Unsave' : 'Save'}
+              </Button>
+              <Button
+                size='small'
+                testID={isPinned ? 'unpinBtn' : 'pinBtn'}
+                disabled={isPinPending || isUnpinPending}
+                type={isPinned ? 'secondary' : 'primary'}
+                onPress={onTogglePinned}
+              >
+                {isPinned ? 'Unpin' : 'Pin to home'}
+              </Button>
+            </>
+          )}
+
+          <NativeDropdown
+            testID="headerDropdownBtn"
+            items={dropdownItems}
+            accessibilityLabel={_(msg`More options`)}
+            accessibilityHint="">
+            <Box bg='l1' py='s' px='m' radius='round'>
```

Comment:
We have a lot of ad-hoc re-used UI like this that we should better abstract. But even so, it's easy to create shapes and layouts on the fly